testing/integ-test/src/crossVersionTest/groovy/org/gradle/integtests/TaskTransitiveSubclassingBinaryCompatibilityCrossVersionSpec.groovy

```diff
@@ -118,9 +118,21 @@ class TaskTransitiveSubclassingBinaryCompatibilityCrossVersionSpec extends Cross
                 }
             }
         """
-
         expect:
-        version previous withTasks 'publish' inDirectory(file("plugin")) run()
-        version current requireDaemon() requireIsolatedDaemons() withTasks 'sofExec' run()
+        version(previous)
+            .withTasks( 'publish')
+            .inDirectory(file("plugin"))
+            .run()
+
+        version(current)
+            .requireDaemon()
+            .requireIsolatedDaemons()
+            .expectDocumentedDeprecationWarning(
+                "In plugin 'com.example.plugin' type 'CustomJavaExec' method 'setArgs' overrides an accessor of a property annotated with @ToBeReplacedByLazyProperty. " +
+                "This behavior has been deprecated. This behavior is scheduled to be removed in Gradle 9.0. " +
+                "Execution optimizations are disabled to ensure correctness. " +
+                "For more information, please refer to https://docs.gradle.org/current/userguide/validation_problems.html#property_accessor_must_not_be_overridden in the Gradle documentation.")
+            .withTasks('sofExec')
+            .run()
```

Comment:
💭 The test here actually overrides `JavaExec.args`, so the deprecation warning is correct.

I don't know why I had to change the syntax like this, but if I just add `version current expectDocumentedDeprecationWarning(...)`, then the expectation gets ignored. 🤷

Reply:
Do we have some caching for metadata store? 🤔

Reply:
We have, but we are running the second build with a different version of Gradle, using an isolated daemon.

But I think the issue is with the test infrastructure instead; when I checked for the expected warnings, the one I added was simply not there.